commit 7ffcd44d7e6c46704f8af0321d9314cd26e0e18a
Reflog: HEAD@{0} (FullwayWang <fullwaywang@tencent.com>)
Reflog message: reset: moving to 7ffcd44d7e6c46704f8af0321d9314cd26e0e18a
Author: Zhipeng Xie <xiezhipeng1@huawei.com>
Date:   Tue Aug 20 16:33:06 2019 +0800

    Fix memory leak in xmlSchemaValidateStream
    
    When ctxt->schema is NULL, xmlSchemaSAXPlug->xmlSchemaPreRun
    alloc a new schema for ctxt->schema and set vctxt->xsiAssemble
    to 1. Then xmlSchemaVStart->xmlSchemaPreRun initialize
    vctxt->xsiAssemble to 0 again which cause the alloced schema
    can not be freed anymore.
    
    Found with libFuzzer.
    
    Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>

diff --git a/xmlschemas.c b/xmlschemas.c
index 301c8449..39d92182 100644
--- a/xmlschemas.c
+++ b/xmlschemas.c
@@ -28090,7 +28090,6 @@ xmlSchemaPreRun(xmlSchemaValidCtxtPtr vctxt) {
     vctxt->nberrors = 0;
     vctxt->depth = -1;
     vctxt->skipDepth = -1;
-    vctxt->xsiAssemble = 0;
     vctxt->hasKeyrefs = 0;
 #ifdef ENABLE_IDC_NODE_TABLES_TEST
     vctxt->createIDCNodeTables = 1;
